torch/_higher_order_ops/invoke_quant.py

```diff
@@ -0,0 +1,87 @@
+# mypy: allow-untyped-decorators
+# mypy: allow-untyped-defs
+import dataclasses
+
+import torch
+import torch.utils._pytree as pytree
+from torch._C import DispatchKey
+from torch._ops import HigherOrderOperator
+from torch._subclasses import FakeTensorMode
+from torch.fx.graph_module import GraphModule
+
+
+class InvokeQuantHOP(HigherOrderOperator):
+    def __init__(self) -> None:
+        super().__init__("invoke_quant")
+
+    def __call__(
+        self,
+        subgraph: GraphModule,
+        *operands,
+        scheme=None,
+    ):
+        return super().__call__(subgraph, *operands)
+
+
+invoke_quant = InvokeQuantHOP()
+
+
+@invoke_quant.py_impl(FakeTensorMode)
+def _(mode, subgraph, *operands, schema=None):
+    return subgraph(*operands)
+
+
+@invoke_quant.py_impl(DispatchKey.CompositeExplicitAutograd)
+def _(subgraph, *operands, schema=None):
+    from torch.utils._python_dispatch import _get_current_dispatch_mode
+
+    mode = _get_current_dispatch_mode()
+    assert mode is None, "Mode should never be enabled for CPU/CUDA key"
+    return subgraph(*operands)
+
+
+@invoke_quant.py_impl(DispatchKey.Autograd)
+def _(subgraph, *operands, schema=None):
+    # invoke_quant should only be invoked post autograd
+    assert not torch.is_grad_enabled() or pytree.tree_all_only(
+        torch.Tensor,
+        lambda t: not t.requires_grad,  # type: ignore[union-attr]
+        operands,
+    )
+    with torch._C._AutoDispatchBelowAutograd():
+        return subgraph(*operands)
+
+
+@dataclasses.dataclass(frozen=True)
+class InvokeQuant:
+    """
+    Invoke a quantization function that will be preserved as a single operator. Preservation
```

Comment:
can you give some examples of quantization function here? are you referring exclusively to dequantize ops? or does it mean any quantization related functions like quantize_affine or quantized kernel as well?

I'm wondering if the higher order op has to mention quant in the name or it can be more general

Reply:
Uh, maybe @drisspg can help with some fp8 prologue scaling functions we want fused. The `uint4x2_mixed_mm` added [here](https://github.com/pytorch/pytorch/pull/106516) is another example that @HDCharles added. 

Yea, it does not specifically have to be quant. The general thing here is about :

a) preservation as a top level op and scheme tagging for special cased lowerings/pattern matching 
b) specific inductor behaviors. 

Some of the patterns seem pretty specific to quant/dequant. 

Specifically I was envisioning: 
- codegen_low_precision
- forcing fusion to mm (both as prologue and epilogue) / autotuning when not max-autotune

maybe there others that come up, not sure.

Reply:
What would a different name be ? And what are the other use cases you're envisioning ?

Reply:
I don't have new use cases.

For naming, just feel mentioning "quant" in a higher order op is a bit weird, if this is the best name we have now that's fine too, something for consideration:
invoke_undecomposed_op / invoke_high_level_op